scripts/packages.mjs

```diff
@@ -12,11 +15,76 @@ export const packageDirsNpmTag = [
 
 export const packageDirsSnyk = ['headless', 'atomic'];
 
+/**
+ * @typedef PackageDefinition
+ * @property {string} name
+ * @property {string} version
+ * @property {string} packageDir Relative to `/packages` (e.g., `atomic-angular/projects/atomic-angular`).
+ */
+
 /**
  * @param {string} fullPath
- * @returns {{name: string, version: string}}
+ * @returns {import('@lerna/package').RawManifest}
  */
 export function getPackageFromPath(fullPath) {
-  const {name, version} = JSON.parse(readFileSync(fullPath).toString());
-  return {name, version};
+  return JSON.parse(readFileSync(fullPath).toString());
+}
+
+/**
+ * @param {string} packageDir E.g.: `samples/some-package`
+ * @returns {PackageDefinition}
+ */
+export function getPackageDefinitionFromPackageDir(packageDir) {
+  const fullPath = resolve('.', 'packages', packageDir, 'package.json');
+  if (!existsSync(fullPath)) {
+    throw `Could not find package at ${fullPath}.`;
+  }
+  const {name, version} = getPackageFromPath(fullPath);
+  return {name, version, packageDir};
+}
+
+/**
+ * @param {string} name
+ */
+export function getPackageDefinitionFromPackageName(name) {
+  return packageDirsNpmTag
+    .map((packageDir) => getPackageDefinitionFromPackageDir(packageDir))
+    .find((packageDef) => packageDef.name === name);
+}
+
+/**
+ * @param {string} packageName
+ * @param {string} newVersion
+ * @param {string[]} depdendenciesPackageDirs E.g.: [`samples/some-package`]
+ */
+export function updatePackageVersion(
+  packageName,
+  newVersion,
+  depdendenciesPackageDirs
+) {
+  depdendenciesPackageDirs.forEach((packageDir) => {
+    const fullPath = resolve('.', 'packages', packageDir, 'package.json');
+    const originalContentAsText = readFileSync(fullPath).toString();
+    const {indent} = detectIndent(originalContentAsText);
+    /** @type {import('@lerna/package').RawManifest} */
+    const manifest = JSON.parse(originalContentAsText);
+
+    if (manifest.name === packageName) {
+      manifest.version = newVersion;
+    } else {
+      if (packageName in (manifest.dependencies || {})) {
+        manifest.dependencies[packageName] = newVersion;
+      }
+      if (packageName in (manifest.devDependencies || {})) {
+        manifest.devDependencies[packageName] = newVersion;
+      }
+      if (packageName in (manifest.peerDependencies || {})) {
+        manifest.peerDependencies[packageName] = newVersion;
+      }
+    }
+    writeFileSync(
+      fullPath,
+      JSON.stringify(manifest, undefined, indent || '  ')
+    );
+  });
```

Comment:
- [ ] try refactoring CDX method + this one into semantic-monorepo-tools.

Reply:
Going to skip this for now, since semantic monorepo tools use the topology

Reply:
u mean CLI I guess but Ok!